### PR TITLE
Improve adherence to TAL coding standards

### PR DESCRIPTION
**Pull request by tsadlerBBC**
Running the TAL codebase against the jshint config found in [TV Application Layer Coding Standards & Guidelines](http://fmtvp.github.io/tal/other/codingstyle.html) will find a multitude of errors.

We have been going through and fixing these on the [jshint branch](https://github.com/fmtvp/tal/tree/jshint) of TAL.

In the interest of not having long lived branches, this pull request is to merge all jshint fixes to date.

### CODE REVIEW

**Comment by tsadlerBBC** [comment]
Group code review found some problems - being dealt with by myself and @rthompson1991.

**Comment by tsadlerBBC** [comment]
@subsidel @jmalk @dhurrell 

@rthompson1991 and I have addressed the issues raised in Wednesday's group code review. Could you please re-review?

Thanks,

Tom

**Comment by subsidel** [comment]
Looks alright from a fairly quick look

File: static/script/widgets/horizontalcarousel.js [hunk 1]
```diff
@@ -19,466 +19,459 @@
...
+            _onDataBound: function (/*evt*/) {
+                var application = this.getCurrentApplication();
+                if (!application) {
+                    // application has been destroyed, abort
+                    return;
+                }
+                var device = application.getDevice();
+
+                if (this._childWidgetOrder.length > 0) {
+                    // TODO: tidy up
```
Comment (dhurrell) [inline static/script/widgets/horizontalcarousel.js]: I assume this TODO relates to the anonymous function that was underneath it but has now been removed, so it should also be removed.

**Comment by dhurrell** [comment]
Looks good to me, apart from the one line comment I've made. Happy for this to be pulled in!

**Comment by tsadlerBBC** [comment]
Thanks guys. I've now removed the redundant TODO.

**Pull request merged**